src/main/java/com/slack/eithernet/Extensions.kt

```diff
@@ -34,6 +34,34 @@ public inline fun <T : Any, E : Any> ApiResult<T, E>.successOrElse(
     is ApiResult.Failure -> defaultValue(this)
   }
 
+/**
+ * If [ApiResult.Success], returns the underlying [T] value. Otherwise, calls [body] with the
+ * failure.
+ */
+public inline fun <T : Any, E : Any> ApiResult<T, E>.successOrNothing(
+  body: (ApiResult.Failure<E>) -> Nothing
```

Comment:
The use of `Nothing` here implies that `body` will never return (ie. it will throw) but there's no way to ensure that happens. Returning `Unit` would seem appropriate, but renaming the method to `successOrUnit` doesn't have the same ring

Reply:
The use of `Nothing` ensures that the body either throws or returns: https://www.zacsweers.dev/its-nothing/

Using `Unit` would actually not accomplish the goal of this function :)

Reply:
Combining `Nothing` with an `inline` function and non-local returns is a cool pattern 👍 . 

I wonder if we should document this somehow. Like me, I expect many/most will probably have to refresh their memory about what exactly `Nothing` is for and come away with only half the story

Reply:
Does the type system not help make that clear here? If you tried to just use `Unit` or anything other than throw/return, this would fail to compile.

Reply:
Sorry, I wasn't clear. I'm not arguing for using `Unit` or any other return type. I'm suggesting that someone unfamiliar with `Nothing` will read the docs and see that they are expected to throw and stop there. They are likely to miss the fact that a non-local return (made possible by the use of an `inline` function) is also a valid option.

I was trying to suggest that we document this a bit better to expose that supported usage.

Reply:
do you have a suggestion? `Nothing` is a pretty standard language feature of Kotlin so I'm not sure what exactly to say other than refer to its doc

Reply:
If it's not clear we could expand the doc to be along the line of;
 `Otherwise, calls [body] with the failure, which should throw or return.`

Reply:
Done

Reply:
This works! Thank you both